source/adapters/cuda/command_buffer.cpp

```diff
@@ -401,6 +401,7 @@ urCommandBufferReleaseExp(ur_exp_command_buffer_handle_t hCommandBuffer) {
 
 UR_APIEXPORT ur_result_t UR_APICALL
 urCommandBufferFinalizeExp(ur_exp_command_buffer_handle_t hCommandBuffer) {
+  UR_ASSERT(!hCommandBuffer->IsFinalized, UR_RESULT_ERROR_INVALID_OPERATION);
```

Comment:
We might be able to check if the `CudaGraphExec` member is not `nullptr`, rather than needing to add a new member variable for `IsFinalized`. i.e `UR_ASSERT(hCommandBuffer->CudaGraphExec == nullptr, UR_RESULT_ERROR_INVALID_OPERATION);`

Reply:
Aha sorry I thought it was alive before this point. Yes that sounds good to me.